### PR TITLE
Multiple domains

### PR DESCRIPTION
**Pull request by davidmir**
I implemented support to allow multiple public tenants (eg domain1.com, domain2.com) https://github.com/bernardopires/django-tenant-schemas/issues/284. 
This can be enabled by flag UNIQUE_PUBLIC_SCHEMA:
* False: schema_name is not unique
* True: schema_name is unique

I've also added a feature that allows to create a list of tenants reserved names by flag NOT_ALLOWED_TENANT_NAMES. 

It would be good to join these features by updating issues in the future.

### CODE REVIEW

File: tenant_schemas/postgresql_backend/base.py [hunk 1]
```diff
@@ -31,8 +31,21 @@ def _check_identifier(identifier):
         raise ValidationError("Invalid string used for the identifier.")
 
 
+def _allowed_schema_name(name):
```
Comment (bernardopires) [inline tenant_schemas/postgresql_backend/base.py]: Please remove anything related to NOT_ALLOWED_TENANT_NAMES.

File: tenant_schemas/__init__.py [hunk 2]
```diff
@@ -42,4 +42,4 @@
     if invalid_schemas:
         raise ImproperlyConfigured(
             "Do not include tenant schemas (%s) on PG_EXTRA_SEARCH_PATHS."
-            % list(invalid_schemas))
+            % list(invalid_schemas))
```
Comment (bernardopires) [inline tenant_schemas/__init__.py]: Please remove this file from the changelist

File: tenant_schemas/models.py [hunk 3]
```diff
@@ -49,7 +51,16 @@ def save(self, verbosity=1, *args, **kwargs):
 
         if is_new and self.auto_create_schema:
             try:
-                self.create_schema(check_if_exists=True, verbosity=verbosity)
+                created = self.create_schema(check_if_exists=True, verbosity=verbosity)
```
Comment (bernardopires) [inline tenant_schemas/models.py]: Is a file missing here? It seems like you changed something inside ``create_schema`` but I don't see it in the changelist.

File: tenant_schemas/utils.py [hunk 4]
```diff
@@ -108,3 +108,16 @@ def app_labels(apps_list):
     Returns a list of app labels of the given apps_list
     """
     return [app.split('.')[-1] for app in apps_list]
+
+
+def unique_public_schema():
+    """
+    Allow multiple domains (e.g. *.domain1.com, *.domain2.com)
+    """
+    if hasattr(settings, 'UNIQUE_PUBLIC_SCHEMA'):
```
Comment (bernardopires) [inline tenant_schemas/utils.py]: Please rename setting to ``SCHEMA_NAME_MUST_BE_UNIQUE`` and the function to ``schema_name_must_be_unique()``.

**Comment by bernardopires** [comment]
Pretty useful feature. Please also add it to the documentation.

File: tenant_schemas/models.py [hunk 5]
```diff
@@ -28,7 +30,7 @@ class TenantMixin(models.Model):
     """
 
     domain_url = models.CharField(max_length=128, unique=True)
-    schema_name = models.CharField(max_length=63, unique=True,
+    schema_name = models.CharField(max_length=63, unique=unique_public_schema(),
```
Comment (goodtune) [inline tenant_schemas/models.py]: How will this work with migrations? If the uniqueness constraint is dynamically determined, won't there be complaints by Django when you switch the setting value?

How about making `unique=False` and using the `clean` method to ensure uniqueness is as you wish?

**Comment by Leeaandrob** [comment]
Merged on master?

**Pull request closed without merging**